tutos/auth-ssh-avec-cle-publique.md

```diff
@@ -0,0 +1,103 @@
+Connexion SSH à l'aide d'une clé publique
+=========================================
+
+Vous en avez mare de taper à chaque fois votre mot de passe
+lorsque [vous vous connectez en SSH](/services/ssh.md) avec votre ordinateur personnel ?
+
+Ce tuto est fait pour vous ! 👈️
+
+
+Création de la paire de clés
+----------------------------
+
+L'authentification par clé publique nécessite une paire de clé de chiffrement.
+Pour créer ces clés, on utilise la {term}`commande` ci-dessous.
+
+    ssh-keygen
+
+Ensuite, le terminal devrait afficher un truc du genre :
+
+    Generating public/private ed25519 key pair.
+    Enter file in which to save the key (/home/LOCAL_USER/.ssh/id_ed25519):
+
+La première ligne nous confirme que c'est bien une "paire de clé" qui est crée.
+La seconde ligne nous demande de choisir l'emplacement de cette paire.
+Ici `LOCAL_USER` devrait correspondre à votre nom d'utilisateur local,
+qui n'est pas forcément le même que celui sur le serveur CLUB1.
+Entre parenthèse, on voit l'emplacement par défaut qui est suggéré : à l'intérieur du {term}`dossier caché` `.ssh`.
+On vous conseille d'accepter cette proposition en appuyant sur {kbd}`entrée`.
+
+    Enter passphrase (empty for no passphrase): 
+
+> Qu'est ce que c'est que ça une _passphrase_ ?
+> je croyais qu'on n'allait plus avoir de mot de passe à taper ? 🫤
+
+Une phrase de passe, c'est en fait exactement la même chose qu'un mot de passe.
+C'est surtout une façon d'encourager à ce qu'elle soit plus longue et donc plus sécurisée.
+
+Sur Linux, les clés seront automatiquement déverrouillées lors de l'ouverture d'une session,
+il ne faut donc pas hésiter à choisir une longue phrase de passe !
```

Comment:
Il me semble que quand tu as Gnome ou KDE l'invite de la passphrase va effectivement proposer de l'enregistrer, mais si on veut être sûr que c'est le cas il vaut mieux ajouter de lancer la commande `ssh-add ~/.ssh/id_ed25519` à un moment donné. Et peut-être indiquer que si la commande échoue alors la passphrase sera demandée à chaque utilisation.

Voir aussi : https://docs.github.com/fr/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Reply:
ah oué intéressant le truc des agents !!